eth_keys.nim

```diff
@@ -0,0 +1,137 @@
+#
+# Nim Ethereum Keys (nim-eth-keys)
+# Copyright (c) 2018 Status Research & Development GmbH
+# Licensed under either of
+# - Apache License, version 2.0, (LICENSE-APACHEv2)
+# - MIT license (LICENSE-MIT)
+#
+
+import nimcrypto/hash, nimcrypto/keccak
+
+{.deadCodeElim:on.}
+
+type
+  EthKeysStatus* = enum
+    Success,  ## Operation was successful
+    Error     ## Operation failed
+
+  EthKeysException* = object of Exception
+    ## Exception generated by this module
+
+when not defined(native):
+  include eth_keys/libsecp256k1
```

Comment:
I've always found includes to be hard to understand - it means that the code in the included file must take into consideration the context in which it's being included - it can no longer be understood on its own

Reply:
So in current situation i have introduced procedure `shutdownLibsecp256k1()` which i dont want to be public, and main module `eth_keys` can use it to properly implement `shutdown()` procedure.

Reply:
In that case you will have to cherry-pick what to export like what is done in master: https://github.com/status-im/nim-eth-keys/blob/f22ae0a827f74ddad0e9c5ce39b64e9f93453f9f/src/eth_keys.nim#L15

Reply:
You can't export something you can't import...

Reply:
I'm curious, what's the reasoning to prefer separate files in that case? why not just dump the contents of the file inside?

if libsecp256k1 can't be thought of as an independent unit that can be used on its own, with a public api that "makes sense" and is completely usable, seems like there are some concepts in here that are not being separated properly